doc/tutorials/content/sources/iterative_closest_point/iterative_closest_point.cpp

```diff
@@ -14,26 +14,32 @@ int
   cloud_in->height   = 1;
   cloud_in->is_dense = false;
   cloud_in->points.resize (cloud_in->width * cloud_in->height);
-  for (std::size_t i = 0; i < cloud_in->points.size (); ++i)
+  
+  for (auto& point : cloud_in->points)
   {
-    cloud_in->points[i].x = 1024 * rand () / (RAND_MAX + 1.0f);
-    cloud_in->points[i].y = 1024 * rand () / (RAND_MAX + 1.0f);
-    cloud_in->points[i].z = 1024 * rand () / (RAND_MAX + 1.0f);
+    point.x = 1024 * rand() / (RAND_MAX + 1.0f);
+    point.y = 1024 * rand() / (RAND_MAX + 1.0f);
+    point.z = 1024 * rand() / (RAND_MAX + 1.0f);
   }
+  
   std::cout << "Saved " << cloud_in->points.size () << " data points to input:"
       << std::endl;
-  for (std::size_t i = 0; i < cloud_in->points.size (); ++i) std::cout << "    " <<
-      cloud_in->points[i].x << " " << cloud_in->points[i].y << " " <<
-      cloud_in->points[i].z << std::endl;
+      
+  for (auto& point : cloud_in->points)
+    std::cout << "    " << point.x << " " << point.y << " " << point.z << std::endl;
```

Comment:
```cpp
for (const auto& point : *cloud_in)
  std::cout << point << std::endl;
```
should be sufficient. The operator exists for `PointXYZ`.
https://github.com/PointCloudLibrary/pcl/blob/3f0114eb942e7a1688917587e9be1db51a57cb7e/common/src/point_types.cpp#L41-L46